src/main/java/edu/ucsb/cs156/frontiers/controllers/TeamsController.java

```diff
@@ -0,0 +1,355 @@
+   * represents a relationship between a team and its members.
+   *
+   * @param courseId the unique identifier of the course for which team mappings are retrieved
+   * @return an iterable collection of {@code TeamMemberMapping} objects representing the mappings
+   */
+  @Operation(summary = "List the mapping of Roster Students to Teams")
+  @PreAuthorize("@CourseSecurity.hasManagePermissions(#root, #courseId)")
+  @GetMapping("/mapping")
+  public Iterable<TeamMemberMapping> teamMemberMapping(@RequestParam Long courseId) {
+    List<Team> teams =
+        courseRepository
+            .findById(courseId)
+            .orElseThrow(() -> new EntityNotFoundException(Course.class, courseId))
+            .getTeams();
+    List<TeamMemberMapping> mappings = new ArrayList<>();
+    for (Team team : teams) {
+      for (TeamMember member : team.getTeamMembers()) {
+        mappings.add(TeamMemberMapping.from(member));
+      }
+    }
+    return mappings;
+  }
+
+  /**
+   * This method returns a single team by its id
+   *
+   * @param id the id of the team
+   * @return the team
+   */
+  @Operation(summary = "Get a single team")
+  @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_INSTRUCTOR')")
+  @GetMapping("")
+  public Team getTeamById(@Parameter(name = "id") @RequestParam Long id) {
+    Team team =
+        teamRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(Team.class, id));
+    return team;
+  }
+
+  /**
+   * This method deletes a team by its id
+   *
+   * @param id the id of the team to delete
+   * @return a message indicating the team was deleted
+   */
+  @Operation(summary = "Delete a team")
+  @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_INSTRUCTOR')")
+  @DeleteMapping("")
+  public Object deleteTeam(@Parameter(name = "id") @RequestParam Long id) {
+    Team team =
+        teamRepository.findById(id).orElseThrow(() -> new EntityNotFoundException(Team.class, id));
+    teamRepository.delete(team);
+    return genericMessage("Team with id %s deleted".formatted(id));
+  }
+
+  /**
+   * This method adds a roster student as a team member
+   *
+   * @param teamId the ID of the team
+   * @param rosterStudentId the ID of the roster student to add
+   * @return the created team member
+   */
+  @Operation(summary = "Add a roster student to a team")
+  @PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_INSTRUCTOR')")
+  @PostMapping("/addMember")
+  public TeamMember addTeamMember(
+      @Parameter(name = "teamId") @RequestParam Long teamId,
+      @Parameter(name = "rosterStudentId") @RequestParam Long rosterStudentId) {
+
+    Team team =
+        teamRepository
+            .findById(teamId)
+            .orElseThrow(() -> new EntityNotFoundException(Team.class, teamId));
+
+    RosterStudent rosterStudent =
+        rosterStudentRepository
+            .findById(rosterStudentId)
+            .orElseThrow(() -> new EntityNotFoundException(RosterStudent.class, rosterStudentId));
+
```

Comment:
Here, we should add a few checks:

1. Does the user have authority to muck around with this course?  Right now, we are only checking  `@PreAuthorize("hasRole('ROLE_ADMIN') || hasRole('ROLE_INSTRUCTOR')")` but we should probably be using the fancy new access control to ensure that the user has access to the course.   That means making the user of the endpoint submit the course id, but honestly, they should have that available.

2. Add a check that the roster student belongs to that course
3. Add a check that the team belongs to that course.